test/service/test_logging.py

```diff
@@ -0,0 +1,65 @@
+import json
+import re
+import requests
+
+from service import WebServiceTestCase
+
+from azul.chalice import log
+from azul.logging import configure_test_logging
+
+
+def setUpModule():
+    configure_test_logging()
+
+
+class TestRequestResponse(WebServiceTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls._setup_indices()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._teardown_indices()
+        super().tearDownClass()
+
+    def _get_request_logs(self, level, endpoint, params={}):
+        with self.assertLogs(logger=log, level=level) as logs:
+            url = self.base_url + endpoint
+            requests.get(url, params=params)
+            return logs.output
+
+    @classmethod
+    def _valid_json(cls, json_string):
+        try:
+            json.loads(json_string)
+        except ValueError:
+            return False
+        return True
+
+    def test_request_logs_debug_off(self):
+        logs = self._get_request_logs(level='INFO', endpoint='/health')
+        log_started_count = sum([1 for msg in logs if re.search(r'^INFO.*Received GET request to \'/health\'', msg)])
```

Comment:
Whenever you enter the backslash key, an alarm bell should go off and you should ask yourself if you can avoid the backslash. In this case, I think you can use `r""`.

Reply:
Rather convoluted. Please learn the distinction between a generator expression and a list comprehension. Study the syntactic shortcut that allows you to omit parentheses around a generator expression when it is the sole argument to another function. Also, `sum([1 for x in xs if x …]) is the same as `len([x in xs if x…])` if I'm not mistaken. The sum(1 for x in xs) is only needed for iterables, like generator expressions, that don't have an inherent length.

Reply:
Between the two options
```
count = sum(1 for log in logs if 'foo' in log)
count = len([log for log in logs if 'foo' in log])
```
the `sum` approach seems more readable IMHO for a one liner list item counter, and has (as you pointed out) has the benefit of being able to use a generator saving memory needed to build a temporary list for `len`.
If you don't approve of stuffing all this into a one liner however I can unpack it to a `for` loop